clang/lib/Sema/SemaDecl.cpp

```diff
@@ -16069,9 +16069,9 @@ Decl *Sema::ActOnFinishFunctionBody(Decl *dcl, Stmt *Body,
       FD->setInvalidDecl();
     } else if (Body) {
       StmtResult SR = SYCL().BuildSYCLKernelCallStmt(FD, Body);
-      if (SR.isInvalid())
-        return nullptr;
-      Body = SR.get();
+      if (!SR.isInvalid()) {
+        Body = SR.get();
+      }
```

Comment:
Glad you asked, I got focused on finishing the patch and forgot to get at the bottom of it.

That's what I did initially, but then the parser hit an assert when trying to parse the function. Looking at this now, I think it is because it bypasses `ExitFunctionBodyRAII` which is meant to pop the eval context upon the function return. I change to this mainly because it is what is done for the cases above and, as you said, an error is emitted so we aren't going to miss compile.

Reply:
made a change to allow early exit